lib/util.js

```diff
@@ -424,21 +438,51 @@ function _fetch_rss_page (feed, page, items, all_items, max_items, oldest_date)
         // number of items, cease crawling
         const done_crawling = (limited_items.length < feed_json.items.length);
 
-        if ((is_paginated || is_array) && !done_crawling) {
-            const new_all_items = new Set(function* () { yield* all_items; yield* new_items.map(item => item.link); }());
+        const new_all_items = new Set(function* () { yield* all_items; yield* new_items.map(item => item.link); }());
+
+        if ((typeof feed_obj !== 'function') && _is_wordpress_feed(feed_json)) {
+            logger.debug('Wordpress feed found, adding pagination');
+            is_paginated = true;
+            const new_feed = _do_create_wordpress(feed_obj);
+            if (Array.isArray(feed)) {
```

Comment:
We call `isArray()` many times in this function; I would prefer if we factored that out in a separate variable.